models/refugee.py

```diff
@@ -0,0 +1,118 @@
+from datetime import date, datetime
+
+class Refugee:
+    """
+    A class to represent a refugee family.
+    """
+    def __init__(self,
+                 firstname: str,
+                 lastname:str,
+                 camp: str,
+                 medicalCondition: str,
+                 numOfFamilyMember: int,
+                 dateOfClosing=None):
+
+        """
+        Initialise a new refugee.
+        :param name of refugee family
+        :param number of family member
+        :param camp of the refugee family locating at
+        :param medical condition of the refugee
+        :param start date of refugee family
+        :param closing date of refugee family
+        """
+
+        self.name = self.__checkAndFormatName(firstname, lastname)
+        self.numOfFamilyMember = self.__checkNumOfFamilyMember(numOfFamilyMember)
+        self.camp = camp
+        self.medicalCondition = medicalCondition
+        self.dateOfCreation = datetime.today().date()
+        self.dateOfClosing = self.__checkDateOfClosing(dateOfClosing)
+
+
+    def __checkAndFormatName(self, firstname, lastname):
+        """
+        format name to a new name
+        """
+        if not isinstance(firstname, str) or not isinstance(lastname, str):
+            raise self.InvalidNameException()
+        elif not firstname.isalpha() or not lastname.isalpha():
+            raise self.InvalidNameException()
+        else:
+            fullname = firstname + " " + lastname
+            return fullname
+        #error handling if invalid name is inputed
+
+    def __checkNumOfFamilyMember(self, numOfFamilyMember):
+        """
+        format number of family member
+        """
+        if not isinstance(numOfFamilyMember, int) or numOfFamilyMember < 0:
+            raise self.InvalidNumOfFamilyMemberException()
+        return numOfFamilyMember + 1
+
+    def formatCamp(self, newCamp):
+        """
+        format identification camp of refugee
+        """
+        if newCamp not in camp.Camp.camps:
+            raise self.InvalidCampException(newCamp)
+        self.camp = newCamp
+
+    def formatMedicalCondition(self, medicalCondition): #medical condition types provided or a description?
```

Comment:
General question: for medical condition, is this the medical condition of the main member of the family or the whole family? If it is the former then having types provided makes sense but if it is the latter then having a description would be better.